presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -264,9 +276,35 @@ public ColumnHandle getUpdateRowIdColumnHandle(ConnectorSession session, Connect
     }
 
     @Override
-    public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
+    public boolean supportsMetadataDelete(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> tableLayoutHandle)
+    {
+        return tableLayoutHandle.map(x -> ((IcebergTableLayoutHandle) x).getTupleDomain().equals(TupleDomain.all())).orElse(true);
+    }
+
+    @Override
+    public OptionalLong metadataDelete(ConnectorSession session, ConnectorTableHandle tableHandle, ConnectorTableLayoutHandle tableLayoutHandle)
     {
-        throw new PrestoException(NOT_SUPPORTED, "This connector only supports delete where one or more partitions are deleted entirely");
+        IcebergTableLayoutHandle layoutHandle = (IcebergTableLayoutHandle) tableLayoutHandle;
+        if (!layoutHandle.getTupleDomain().equals(TupleDomain.all())) {
+            throw new PrestoException(INVALID_ARGUMENTS, "iceberg metadata delete doesn't support predicates");
```

Comment:
Is this reachable given the check in the previous function?

Reply:
I couldn't generate a test case where this exception would be thrown. However while I assumed that `metadataDelete` would only be called when `supportsMetadataDelete` returns true, I don't actually know the contract between this method and `supportsMetadataDelete`. Is it ever possible it could change in the future. I added this conditional in as a defensive measure in case this was ever called. I suppose instead I could change this condition to just `if (!supportsMetadataDelete(session, tableHandle, Optional.of(layoutHandle))) {` instead to consolidate the logic in the case we want to keep this defensive check

If you think it is unnecessary, then I am not really against removing it

Reply:
I think it is low overhead and ok to leave it. Yes - consolidating the two checks makes sense

Reply:
fixed